gradle-plugin/it/src/test/java/org/eclipse/jkube/gradle/plugin/tests/DependencyResourcesIT.java

```diff
@@ -63,4 +68,20 @@ public void k8sResource_whenRunWithReplicas_generatesK8sManifestsIncludingDepend
         .contains("Adding revision history limit to 2")
         .contains("validating");
   }
+
+  private static String[] addCleanArgument(String... arguments) {
+    Builder<String> builder = Stream.builder();
+    /*
+     * For some weird reason, depending on the order of execution of the tests, the first running test locks the entire
+     * build folder even after it has finished running and, as a result, the next test can't clean such folder and it
+     * fails.
+     * Considering that the tests operate on different resources, it should be safe to skip the clean step completely.
+     */
+    if (!SystemUtils.IS_OS_WINDOWS) {
+      builder.add("clean");
+    }
+    Arrays.stream(arguments).forEach(builder);
+
+    return builder.build().toArray(String[]::new);
+  }
```

Comment:
We'll dig a little bit more on this.

Reply:
I did some researches about this, and it seems like that the Gradle daemon stays up even when the Gradle execution completes (well, it's a daemon after all), and this may be the issue.
I tried to use the [--stop](https://docs.gradle.org/current/userguide/gradle_daemon.html#sec:stopping_an_existing_daemon) flag to stop it, but it's not recognized for some reason (maybe it's not supported in version 6.9 you're using) and I don't want to hack the user's home with the `gradle.properties` file.

Reply:
I hope this is not the issue because we shouldn't stop the daemon between executions (we need to take advantage of that to make the tests quick).
I'll try to check this part on my machine later this week.

Reply:
Wouldn't then be better to start it from Maven in the `pre-integration-test` phase and stop it in the `post-integration-phase` then? The real question is then how to connect/use it from the tests...

Reply:
We should just not tamper with the gradle executable and rely on the wrapper to work accordingly on the user's machine.

Just like you wouldn't normally kill your gradle daemons after a `gradle build` (or whatever) execution.

We could always use the `--no-daemon` command-line flag, but again, this would affect test execution performance. The idea would be to understand what's actually going on and why is the directory->folder locked.